pycode/memilio-epidata/memilio/epidata/getCaseData.py

```diff
@@ -0,0 +1,676 @@
+        gd.write_dataframe(gbNF_cs, directory, prefix + filename, file_format)
+
+    ############## Data for states all ages ################
+
+    # NeuerFall: Infected (incl. recovered) over "dateToUse" for every state ("Bundesland"):
+    gbNFst = df.groupby([dateToUse, IdBundesland]).agg({AnzahlFall: sum})
+    gbNFst_cs = gbNFst.groupby(level=1).cumsum().reset_index()
+
+    # output
+    filename = 'infected_state'
+    if rep_date:
+        filename_orig = filename + '_repdate'
+    else:
+        filename_orig = filename
+    gd.write_dataframe(gbNFst_cs, directory,
+                       prefix + filename_orig, file_format)
+    if impute_dates or moving_average > 0:
+        gbNFst_cs = modifyDataframeSeries.impute_and_reduce_df(
+            gbNFst_cs,
+            {dd.EngEng["idState"]: [k for k, v in dd.State.items()]},
+            ['Confirmed'],
+            impute='forward', moving_average=moving_average)
+        filename = gd.append_filename(filename, impute_dates, moving_average)
+        if rep_date:
+            filename = filename + '_repdate'
+        gd.write_dataframe(gbNFst_cs, directory,
+                           prefix + filename, file_format)
+
+    # output nested json
+    # gbNFst_cs.groupby(['IdBundesland'], as_index=False) \
+    #            .apply(lambda x: x[[dateToUse,'AnzahlFall']].to_dict('r')) \
+    #            .reset_index().rename(columns={0:'Dates'})\
+    #            .to_json(directory + "gbNF_state_nested.json", orient='records')
+
+    # infected (incl recovered), deaths and recovered together
+
+    gbAllSt = df.groupby([dateToUse, IdBundesland]) \
+        .agg({AnzahlFall: sum, AnzahlTodesfall: sum, AnzahlGenesen: sum})
+    gbAllSt_cs = gbAllSt.groupby(level=1).cumsum().reset_index()
+
+    # output
+    filename = 'all_state'
+    if rep_date:
+        filename_orig = filename + '_repdate'
+    else:
+        filename_orig = filename
+    gd.write_dataframe(gbAllSt_cs, directory,
+                       prefix + filename_orig, file_format)
+    if impute_dates or moving_average > 0:
+        gbAllSt_cs = modifyDataframeSeries.impute_and_reduce_df(
+            gbAllSt_cs,
+            {dd.EngEng["idState"]: [k for k, v in dd.State.items()]},
+            ['Confirmed', 'Deaths', 'Recovered'],
+            impute='forward', moving_average=moving_average)
+        filename = gd.append_filename(filename, impute_dates, moving_average)
+        if rep_date:
+            filename = filename + '_repdate'
+        gd.write_dataframe(gbAllSt_cs, directory,
+                           prefix + filename, file_format)
+
+    ############# Data for counties all ages ######################
+
+    if not split_berlin:
+        df = geoger.merge_df_counties(
+            df, 11000, geoger.CountyMerging[11000],
+            sorting=[dd.EngEng['date']],
+            columns=[dd.EngEng['date'],
+                     dd.EngEng['gender'],
+                     dd.EngEng['idState'],
```

Comment:
I checked this and you are right the result is different. 
In case the county is added to the columns the merging does not sum up the data of the counties merged but overwrites the name and countyid to the ones of the merged county. It only sums up the data for the separated counties of berlin.
In the other case (as it is written now) the data is summed up over the different counties so there is at most one data set for Berlin for the same date, gender and age. I thought this is the way the function was intended to work based on the name. 
Maybe this is also why some shapes in the test have changed.